Failed: ActorsLeakSpec

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/6159/testReport/junit/akka.remote/ActorsLeakSpec/Remoting_must_not_leak_actors/

```
sbt.ForkMain$ForkError: java.lang.AssertionError: received 1 excess messages on WarningFilter(None,Right(Association with remote system),false)
	at akka.testkit.EventFilter.intercept(TestEventListener.scala:116)
	at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$4.apply(ActorsLeakSpec.scala:164)
	at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$4.apply(ActorsLeakSpec.scala:141)
	at scala.collection.immutable.Range.foreach(Range.scala:160)
	at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ActorsLeakSpec.scala:141)
	at akka.remote.ActorsLeakSpec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(ActorsLeakSpec.scala:69)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scal
```

Maybe not same as above?

https://jenkins.akka.io:8498/job/akka-artery-aeron-udp-remote-tests/141

```
[ERROR] [09/20/2019 08:51:12.042] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-21] [akka.remote.Remoting] Association to [akka.trttl.tcp://remote@127.0.1.1:35889] with UID [-400037490] irrecoverably failed. Quarantining address.
java.util.concurrent.TimeoutException: Remote system has been silent for too long. (more than 3 seconds)
	at akka.remote.ReliableDeliverySupervisor$$anonfun$idle$1.applyOrElse(Endpoint.scala:454)
	at akka.actor.Actor.aroundReceive(Actor.scala:532)
	at akka.actor.Actor.aroundReceive$(Actor.scala:530)
	at akka.remote.ReliableDeliverySupervisor.aroundReceive(Endpoint.scala:249)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:573)
	at akka.actor.ActorCell.invoke(ActorCell.scala:543)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
	at akka.dispatch.Mailbox.run(Mailbox.scala:230)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:242)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

[WARN] [09/20/2019 08:51:12.042] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-21] [akka.remote.Remoting] Association to [akka.trttl.tcp://remote@127.0.1.1:35889] having UID [-400037490] is irrecoverably failed. UID is now quarantined and all messages to this UID will be delivered to dead letters. Remote ActorSystem must be restarted to recover from this situation.
[WARN] [09/20/2019 08:51:12.044] [ActorsLeakSpec-akka.actor.default-dispatcher-22] [akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A35889-7] received dead letter from Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A35889-7#113123325]: AttemptSysMsgRedelivery
[WARN] [09/20/2019 08:51:12.382] [ActorsLeakSpec-akka.actor.default-dispatcher-22] [akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34613-9/endpointWriter] received dead letter from Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34613-9/endpointWriter#-368047486]: BackoffTimer
[WARN] [09/20/2019 08:51:16.222] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-8] [akka.remote.Remoting] Association to [akka.trttl.tcp://remote@127.0.1.1:34811] having UID [439382821] is irrecoverably failed. UID is now quarantined and all messages to this UID will be delivered to dead letters. Remote ActorSystem must be restarted to recover from this situation.
[WARN] [09/20/2019 08:51:16.222] [ActorsLeakSpec-akka.actor.default-dispatcher-22] [akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34811-8] received dead letter from Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34811-8#-488219663]: AttemptSysMsgRedelivery
[WARN] [09/20/2019 08:51:17.395] [New I/O boss #267] [NettyTransport(akka://ActorsLeakSpec)] Remote connection to [null] failed with java.net.ConnectException: Connection refused: /127.0.1.1:34613
[WARN] [09/20/2019 08:51:17.396] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-21] [akka.trttl.tcp://ActorsLeakSpec@127.0.1.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34613-9] Association with remote system [akka.trttl.tcp://remote@127.0.1.1:34613] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.trttl.tcp://remote@127.0.1.1:34613]] Caused by: [java.net.ConnectException: Connection refused: /127.0.1.1:34613]
[WARN] [09/20/2019 08:51:22.418] [New I/O boss #267] [NettyTransport(akka://ActorsLeakSpec)] Remote connection to [null] failed with java.net.ConnectException: Connection refused: /127.0.1.1:34613
[WARN] [09/20/2019 08:51:22.419] [ActorsLeakSpec-akka.remote.default-remote-dispatcher-21] [akka.trttl.tcp://ActorsLeakSpec@127.0.1.1:2552/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34613-9] Association with remote system [akka.trttl.tcp://remote@127.0.1.1:34613] has failed, address is now gated for [5000] ms. Reason: [Association failed with [akka.trttl.tcp://remote@127.0.1.1:34613]] Caused by: [java.net.ConnectException: Connection refused: /127.0.1.1:34613]
[info] - must not leak actors *** FAILED *** (31 seconds, 67 milliseconds)
[info]   Expected Set(Actor[akka://ActorsLeakSpec/system/endpointManager#-1671816760], Actor[akka://ActorsLeakSpec/system/transports#1394778779], Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp56#-743046706], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp4#75720964]), but got Set(Actor[akka://ActorsLeakSpec/system/endpointManager#-1671816760], Actor[akka://ActorsLeakSpec/system/transports/throttlermanager.tcp4#75720964], Actor[akka://ActorsLeakSpec/system/endpointManager/reliableEndpointWriter-akka.trttl.tcp%3A%2F%2Fremote%40127.0.1.1%3A34613-9#-1265002418], Actor[akka://ActorsLeakSpec/system/transports/akkaprotocolmanager.trttl.tcp56#-743046706], Actor[akka://ActorsLeakSpec/system/transports#1394778779]) (ActorsLeakSpec.scala:217)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
[info]   at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
[info]   at akka.testkit.AkkaSpec.newAssertionFailedException(AkkaSpec.scala:61)
[info]   at org.scalatest.Assertions.assertResult(Assertions.scala:1025)
[info]   at org.scalatest.Assertions.assertResult$(Assertions.scala:1021)
[info]   at akka.testkit.AkkaSpec.assertResult(AkkaSpec.scala:61)
[info]   at akka.remote.classic.ActorsLeakSpec.$anonfun$new$13(ActorsLeakSpec.scala:217)
[info]   at akka.testkit.TestKitBase.poll$2(TestKit.scala:327)
[info]   at akka.testkit.TestKitBase.awaitAssert(TestKit.scala:344)
[info]   at akka.testkit.TestKitBase.awaitAssert$(TestKit.scala:316)
[info]   at akka.testkit.TestKit.awaitAssert(TestKit.scala:928)
[info]   at akka.remote.classic.ActorsLeakSpec.$anonfun$new$2(ActorsLeakSpec.scala:217)

```

--- a/akka-remote/src/test/scala/akka/remote/classic/ActorsLeakSpec.scala
+++ b/akka-remote/src/test/scala/akka/remote/classic/ActorsLeakSpec.scala
@@ -209,9 +209,7 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
         remoteSystem.terminate()
       }
 
-      EventFilter.warning(pattern = "Association with remote system", occurrences = 1).intercept {
-        Await.ready(remoteSystem.whenTerminated, 10.seconds)
-      }
+      Await.ready(remoteSystem.whenTerminated, 10.seconds)
 
       EventFilter[TimeoutException](occurrences = 1).intercept {}
 

--- a/akka-remote/src/test/scala/akka/remote/classic/ActorsLeakSpec.scala
+++ b/akka-remote/src/test/scala/akka/remote/classic/ActorsLeakSpec.scala
@@ -104,7 +104,7 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
           remoteSystem.terminate()
         }
 
-        Await.ready(remoteSystem.whenTerminated, 10.seconds)
+        Await.result(remoteSystem.whenTerminated, 10.seconds)
       }
 
       // Quarantine an old incarnation case
@@ -147,7 +147,7 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
           remoteSystem.terminate()
         }
 
-        Await.ready(remoteSystem.whenTerminated, 10.seconds)
+        Await.result(remoteSystem.whenTerminated, 10.seconds)
 
       }
 
@@ -170,14 +170,14 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
           probe.expectMsgType[ActorIdentity].ref.nonEmpty should be(true)
 
           // This will make sure that no SHUTDOWN message gets through
-          Await.ready(RARP(system).provider.transport.managementCommand(ForceDisassociate(remoteAddress)), 3.seconds)
+          Await.result(RARP(system).provider.transport.managementCommand(ForceDisassociate(remoteAddress)), 3.seconds)
 
         } finally {
           remoteSystem.terminate()
         }
 
         EventFilter.warning(pattern = "Association with remote system", occurrences = 1).intercept {
-          Await.ready(remoteSystem.whenTerminated, 10.seconds)
+          Await.result(remoteSystem.whenTerminated, 10.seconds)
         }
       }
 
@@ -203,13 +203,13 @@ class ActorsLeakSpec extends AkkaSpec(ActorsLeakSpec.config) with ImplicitSender
         // All system messages has been acked now on this side
 
         // This will make sure that no SHUTDOWN message gets through
-        Await.ready(RARP(system).provider.transport.managementCommand(ForceDisassociate(remoteAddress)), 3.seconds)
+        Await.result(RARP(system).provider.transport.managementCommand(ForceDisassociate(remoteAddress)), 3.seconds)
 
       } finally {
         remoteSystem.terminate()
       }
 
-      Await.ready(remoteSystem.whenTerminated, 10.seconds)
+      Await.result(remoteSystem.whenTerminated, 10.seconds)
 
       EventFilter[TimeoutException](occurrences = 1).intercept {}
 

